test/e2e/prometheus_test.go

```diff
@@ -3905,6 +3905,100 @@ func testPromEnforcedNamespaceLabel(t *testing.T) {
 	}
 }
 
+func testPrometheusCRDValidation(t *testing.T) {
+	t.Parallel()
+	name := "test"
+	replicas := int32(1)
+	commonFields := monitoringv1.CommonPrometheusFields{
+		Replicas: &replicas,
+		Version:  operator.DefaultPrometheusVersion,
+
+		ServiceAccountName: "prometheus",
+		Resources: v1.ResourceRequirements{
+			Requests: v1.ResourceList{
+				v1.ResourceMemory: resource.MustParse("400Mi"),
+			},
+		},
+	}
+
+	tests := []struct {
+		name           string
+		prometheusSpec monitoringv1.PrometheusSpec
+		expectedError  bool
+	}{
+		//
+		// RetentionSize Validation:
+		//
+		{
+			name: "zero-size-without-unit",
+			prometheusSpec: monitoringv1.PrometheusSpec{
+				CommonPrometheusFields: commonFields,
+				RetentionSize:          "0",
+			},
+		},
+		{
+			name: "legacy-unit",
+			prometheusSpec: monitoringv1.PrometheusSpec{
+				CommonPrometheusFields: commonFields,
+				RetentionSize:          "1.5GB",
+			},
+		},
+		{
+			name: "iec-unit",
+			prometheusSpec: monitoringv1.PrometheusSpec{
+				CommonPrometheusFields: commonFields,
+				RetentionSize:          "100MiB",
+			},
+		},
+		{
+			name: "legacy-missing-symbol",
```

Comment:
I had to cut short the suggestion to avoid error while creating namespace since we use this name

``` 
namespace.go:39: failed to create namespace with name allns-y-prometheuscrdvalidation-legacy-unit-with-missing-symbol-r971yd-0: Namespace "allns-y-prometheuscrdvalidation-legacy-unit-with-missing-symbol-r971yd-0" is invalid: [metadata.name: Invalid value: "allns-y-prometheuscrdvalidation-legacy-unit-with-missing-symbol-r971yd-0": must be no more than 63 characters, metadata.labels: Invalid value: "allns-y-prometheuscrdvalidation-legacy-unit-with-missing-symbol-r971yd-0": must be no more than 63 characters]
```